addons/point_of_sale/models/pos_config.py

```diff
@@ -643,12 +643,15 @@ def post_install_pos_localisation(self, companies=False):
             pos_configs = self.search(domain)
             if not pos_configs:
                 self = self.with_company(company)
-                pos_configs = self.env['pos.config'].create({
-                'name': _('Shop'),
-                'company_id': company.id,
-                'module_pos_restaurant': False,
-            })
-            pos_configs.setup_defaults(company)
+                if picking_type_id := self._default_picking_type_id():
+                    pos_configs = self.env['pos.config'].create({
+                    'name': _('Shop'),
+                    'company_id': company.id,
+                    'module_pos_restaurant': False,
+                    'picking_type_id': picking_type_id
```

Comment:
Fix the indentation while you are at it
```suggestion
                        'name': _('Shop'),
                        'company_id': company.id,
                        'module_pos_restaurant': False,
                        'picking_type_id': picking_type_id
```